### PR TITLE
move get device key API from client to crypto

### PR DESCRIPTION
**Pull request by BillCarsonFr**
`MatrixClient` API was exposing two methods that only worked for legacy crypto:
    - `getDeviceEd25519Key`
    - `getDeviceCurve25519Key`

=> These are used in the react-sdk for some functionality (rageshake, sentry, rendez-vous).

I have deprectaded those calls from `MatrixClient` and created a new API in `CryptoApi` (where it belongs):

`getOwnDeviceKeys(): Promise<OwnDeviceKeys>`

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/rust-crypto/rust-crypto.ts [hunk 1]
```diff
@@ -157,6 +157,21 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
...
+        // could be undefined if there is no such algorithm for that device.
+        if (device.curve25519Key && device.ed25519Key) {
+            return {
+                ed25519: device.ed25519Key.toBase64(),
+                curve25519: device.curve25519Key.toBase64(),
+            };
+        }
+        throw new Error("Device keys not found");
+    }
+
```
Comment (richvdh) [inline src/rust-crypto/rust-crypto.ts:174]: Please could you move this underneath the `CryptoApi implementation` comment (line 375).

File: src/rust-crypto/rust-crypto.ts [hunk 2]
```diff
@@ -157,6 +157,21 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
         this.checkKeyBackupAndEnable();
     }
 
+    public async getOwnDeviceKeys(): Promise<OwnDeviceKeys> {
```
Comment (richvdh) [inline src/rust-crypto/rust-crypto.ts:160]:
```suggestion
    /**
     * Implementation of {@link CryptoApi#getOwnDeviceKeys}.
     */
    public async getOwnDeviceKeys(): Promise<OwnDeviceKeys> {
```

**Review by richvdh — COMMENTED** [review]
Looks good, but quite a few nits

**Comment by richvdh** [comment]
coverage check is complaining. Is it possible to test the error case in RustCrypto?

File: src/client.ts [hunk 3]
```diff
@@ -2400,6 +2403,8 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
      *
      * @returns base64-encoded curve25519 key. Null if crypto is
      *    disabled.
+     *
+     * @deprecated Use {@link CryptoApi.getOwnDeviceKeys}
```
[the remark this answers is not in the document]
Comment (BillCarsonFr) [reply src/client.ts:2407]: I hesitated but the problem is that it's not working for rust crypto. So I thought "Use" was better

**Comment by BillCarsonFr** [comment]
> coverage check is complaining. Is it possible to test the error case in RustCrypto?

Don't see how to do it. Would have to simulate when the ed and curve key are not defined. It would be either that there was a crypto store error or that the current device doesn't support ed25519 and curve25519

**Pull request merged**